File: js/charCount.js
exports.charCount = function(string) {
    let strObj = {};	
    string = string.toLowerCase();
    // iterate over string	
    for (let i = 0; i < string.length; i++){
        // if string is a space, remove it	
        if (string[i] === ' '){	
        continue;	
        }
        // if letter isn't in object, initialize it with value of one	
        if (strObj[string[i]] === undefined){	
            strObj[string[i]] = 1;
        // if letter already a key in object, increment it by one	
        } else {	
            strObj[string[i]] = strObj[string[i]] + 1;	
        }	
    }	
    let result = [];
    // push keys/values into array (there's a better way to do this with one command)
    for (key in strObj){
        let val = strObj[key];
        result.push([key, val]);
    }
    // need to determine the proper sort function to get seconday alphabetical sort **********
    result.sort((a,b) => b[1] - a[1]);
    // result.sort(([a, b], [c, d]) => d - b || c - a);
    return result;	
};
console.log(exports.charCount("an apple a day will keep the doctor away"))

File: python/char_count.py
def char_count(string):
  chars = {}
  result = []
  string = string.lower()
  # remove spaces between words
  string = ''.join(string.split(' '))
  for letter in string:
    # retrieve *list* of current keys in dictionary
    keys = chars.keys()
    #check if key exists for current letter
    if letter in keys:
      # if so, increment by one
      chars[letter] += 1
    else:
      # if not, initialize to one
      chars[letter] = 1
  # iterate over keys
  for key in chars:
    # add key/value to result list
    result.append([key, chars[key]])
  # sort result list by key value (not the key, itself)
  # lambda function = first by key value (descending), then by key (ascending)
  result = sorted(result, key = lambda x: (-x[1],x[0]))
  return(result)

#print(char_count("an apple a day will keep the doctor away"))